diff -up scribus-1.5.4/scribus/plugins/import/pdf/importpdf.cpp.omv~ scribus-1.5.4/scribus/plugins/import/pdf/importpdf.cpp
--- scribus-1.5.4/scribus/plugins/import/pdf/importpdf.cpp.omv~	2018-10-11 12:47:45.405315500 +0200
+++ scribus-1.5.4/scribus/plugins/import/pdf/importpdf.cpp	2018-10-11 13:10:49.745005104 +0200
@@ -556,12 +556,10 @@ bool PdfPlug::convert(const QString& fn)
 									}
 									else
 									{
-										GooList *ocgs;
-										int i;
-										ocgs = ocg->getOCGs ();
-										for (i = 0; i < ocgs->getLength (); ++i)
+										const auto &ocgs = ocg->getOCGs ();
+										for (const auto &ocg : ocgs)
 										{
-											OptionalContentGroup *oc = (OptionalContentGroup *)ocgs->get(i);
+											OptionalContentGroup *oc = ocg.second.get();
 											QString ocgName = UnicodeParsedString(oc->getName());
 											if (!ocgNames.contains(ocgName))
 											{
@@ -574,12 +572,10 @@ bool PdfPlug::convert(const QString& fn)
 							}
 							else
 							{
-								GooList *ocgs;
-								int i;
-								ocgs = ocg->getOCGs ();
-								for (i = 0; i < ocgs->getLength (); ++i)
+								const auto &ocgs = ocg->getOCGs ();
+								for (const auto &ocg : ocgs)
 								{
-									OptionalContentGroup *oc = (OptionalContentGroup *)ocgs->get(i);
+									OptionalContentGroup *oc = ocg.second.get();
 									QString ocgName = UnicodeParsedString(oc->getName());
 									if (!ocgNames.contains(ocgName))
 									{
